File: carddata/src/main/java/com/cardemory/carddata/entity/Card.kt
package com.cardemory.carddata.entity

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Card(
    val id: Long = UNDEFINED_ID,
    val cardSetId: Long,
    val title: String,
    val description: String,
    val memoryRank: Double,
    val lastTrainMillis: Long
) : Parcelable {

    constructor() : this(UNDEFINED_ID, -1, "", "", 0.0, 0)

    companion object {
        const val UNDEFINED_ID = -1L
    }
}

File: flutter/cardemory/lib/features/create_card/domain/entities/card.dart
import 'package:equatable/equatable.dart';

class Card extends Equatable {
  static const unknownId = -1;

  final int id;
  final String title;
  final String description;

  const Card(this.title, this.description, {this.id = unknownId});

  @override
  List<Object> get props => [id, title, description];
}
